test/Microsoft.AspNet.Mvc.TagHelpers.Test/CacheTagHelpersTest.cs

```diff
@@ -0,0 +1,621 @@
+                MemoryCache = cache
+            };
+            cacheTagHelper2.ViewContext.HttpContext.Request.Headers["Cookie"] = "cookie1=value1;cookie2=not-value2";
+
+            // Act - 2
+            await cacheTagHelper2.ProcessAsync(tagHelperContext2, tagHelperOutput2);
+
+            // Assert - 1
+            Assert.Null(tagHelperOutput2.PreContent);
+            Assert.Null(tagHelperOutput2.PostContent);
+            Assert.True(tagHelperOutput2.ContentSet);
+            Assert.Equal(childContent2, tagHelperOutput2.Content);
+        }
+
+        [Fact]
+        public void UpdateCacheContext_SetsAbsoluteExpiration_IfExpiresOnIsSet()
+        {
+            // Arrange
+            var expiresOn = DateTimeOffset.UtcNow.AddMinutes(4);
+            var cache = new MemoryCache(new MemoryCacheOptions());
+            var cacheContext = new Mock<ICacheSetContext>();
+            cacheContext.Setup(c => c.SetAbsoluteExpiration(expiresOn))
+                        .Verifiable();
+            var cacheTagHelper = new CacheTagHelper
+            {
+                MemoryCache = cache,
+                ExpiresOn = expiresOn
+            };
+
+            // Act
+            cacheTagHelper.UpdateCacheContext(cacheContext.Object);
+
+            // Assert
+            cacheContext.Verify();
+        }
+
+        [Fact]
+        public void UpdateCacheContext_SetsAbsoluteExpiration_IfExpiresAfterIsSet()
+        {
+            // Arrange
+            var expiresAfter = TimeSpan.FromSeconds(42);
+            var cache = new MemoryCache(new MemoryCacheOptions());
+            var cacheContext = new Mock<ICacheSetContext>();
+            cacheContext.Setup(c => c.SetAbsoluteExpiration(expiresAfter))
+                        .Verifiable();
+            var cacheTagHelper = new CacheTagHelper
+            {
+                MemoryCache = cache,
+                ExpiresAfter = expiresAfter
+            };
+
+            // Act
+            cacheTagHelper.UpdateCacheContext(cacheContext.Object);
+
+            // Assert
+            cacheContext.Verify();
+        }
+
+        [Fact]
+        public void UpdateCacheContext_SetsSlidingExpiration_IfExpiresSlidingIsSet()
+        {
+            // Arrange
+            var expiresSliding = TimeSpan.FromSeconds(37);
+            var cache = new MemoryCache(new MemoryCacheOptions());
+            var cacheContext = new Mock<ICacheSetContext>();
+            cacheContext.Setup(c => c.SetSlidingExpiration(expiresSliding))
+                        .Verifiable();
+            var cacheTagHelper = new CacheTagHelper
+            {
+                MemoryCache = cache,
+                ExpiresSliding = expiresSliding
+            };
+
+            // Act
+            cacheTagHelper.UpdateCacheContext(cacheContext.Object);
+
+            // Assert
+            cacheContext.Verify();
+        }
+
+        [Fact]
+        public void UpdateCacheContext_SetsCachePreservationPriority()
```

Comment:
add a test confirming the `ICacheSetContext` is called multiple times if multiple `Expires*` properties are set